doc/debug-messages.txt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 1996-2021 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
```

Comment:
It should be no more difficult to generate a .dox than a .txt

Reply:
@eduard-bagdasaryan, perhaps you can wrap the whole generated output in `'''` or whatever Doxygen is using for verbatim text? IMO, there is not much value (and there is certainly some harm!) in adding markup to each generated entry. With the suggested one-verbatim-block approach:

* those unicorn admins that use Doxygen documentation when configuring Squid can read preformatted text just fine;
* other admins will not be harmed by Doxygen markup polluting these simple entries on their plain text terminals.

Reply:
Oh you expect people other than dev to read this document? then it should be in the wiki or at worst a list in the config documentation.

Reply:
> Oh you expect people other than dev to read this document?

Yes, of course. Admins need this document to find the IDs of the messages they want to suppress. This document is referenced from squid.conf.documented. Developers may consult it too, of course, but they are not the primary audience.

> then it should be in the wiki or at worst a list in the config documentation.

Why should it be?! The debug-sections.txt document is not.

Reply:
I added into source-maintenance.sh the requested functionality for debug-section.dox generation. 

>  for verbatim text

I enclosed the text into the 'verbatim/endverbatim' Doxygen commands.

Reply:
Please do not forget to commit the new doxygen file, remove the obsolete txt file, and push those changes. It would also be nice to paste Doxygen HTML output to PR so that reviewers can see how the rendered text looks.

Reply:
Done.

Reply:
> > then it should be in the wiki or at worst a list in the config documentation.
> 
> Why should it be?! The debug-sections.txt document is not.

https://wiki.squid-cache.org/KnowledgeBase/DebugSections

Reply:
> Amos: https://wiki.squid-cache.org/KnowledgeBase/DebugSections

That is not a debug-sections.txt document! That is a manual rendering inspired by the debug-sections.txt document.

Needless to say, somebody can create a similar manual rendering of the new debug-messages as well, but we should not encourage (not to mention require) such wasteful actions IMO because such renderings cannot be easily done in a version-specific manner and are all but guaranteed to become stale. It would be much better to (automatically) publish version-specific [links](https://raw.githubusercontent.com/squid-cache/squid/SQUID_4_13/doc/debug-sections.txt) (based on the git repository tag) to the GitHub-hosted text files.

Reply:
> Amos: It should be no more difficult to generate a .dox than a .txt

It (naturally) is, but you got your wish.

FWIW, I think it was a bad idea -- more complex generation code, less readable text, very difficult to produce the final rendering, antiquated formatting, more inconsistencies with debug-sections.txt. This issue is not important enough for me to fight for, but if, after evaluating the result, you decide that you are are OK with plain text, please let me know, and I or Eduard will remove Doxygen wrapping from the PR.